File: Selenium/Activities/Activity5.java
package activity;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;

public class Activity5 {

	public static void main(String[] args) {

		WebDriver driver = new FirefoxDriver();

        driver.get("https://training-support.net/webelements/dynamic-controls");
        System.out.println("Page Title: " + driver.getTitle());

        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));                 
        WebElement checkbox = driver.findElement(By.id("checkbox"));
        driver.findElement(By.xpath("//button[text()='Toggle Checkbox']")).click();
        System.out.println("Checkbox is displayed: " + checkbox.isDisplayed());
        driver.findElement(By.xpath("//button[text()='Toggle Checkbox']")).click();
        System.out.println("Checkbox is displayed: " + checkbox.isDisplayed());
        driver.quit();
    }
}

File: Selenium/Activities/activity5.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


driver = webdriver.Firefox()

try:
    
    driver.get("https://training-support.net/webelements/dynamic-controls")
    driver.maximize_window()

    
    print("Page Title:", driver.title)

    
    wait = WebDriverWait(driver, 10)

    
    checkbox = driver.find_element(By.ID, "checkbox")

    
    driver.find_element(By.XPATH, "//button[text()='Toggle Checkbox']").click()

    
    wait.until(EC.invisibility_of_element_located((By.ID, "checkbox")))
    print("Checkbox is removed.")

    
    driver.find_element(By.XPATH, "//button[text()='Toggle Checkbox']").click()

    
    wait.until(EC.visibility_of_element_located((By.ID, "checkbox")))

    
    checkbox = driver.find_element(By.ID, "checkbox")

    
    print("Checkbox is displayed:", checkbox.is_displayed())

finally:
    
    driver.quit()